agent/transport/transport.go

```diff
@@ -0,0 +1,156 @@
+package transport
+
+import (
+	"io"
+	"sort"
+	"time"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+// A Factory is used to initialize a new transport.
+type Factory func(*zap.Logger, TaskWriter) (io.WriteCloser, error)
```

Comment:
why not NewTransport?

Reply:
NewTransport seems like it should be the name of a method that returns a transport, not the name of a type that is responsible for instantiating a transport.